docs/attributes-registry/messaging.md

```diff
@@ -16,6 +16,7 @@
 | `messaging.destination.temporary` | boolean | A boolean that is true if the message destination is temporary and might not exist anymore after messages are processed. |  |
 | `messaging.destination_publish.anonymous` | boolean | A boolean that is true if the publish message destination is anonymous (could be unnamed or have auto-generated name). |  |
 | `messaging.destination_publish.name` | string | The name of the original destination the message was published to [4] | `MyQueue`; `MyTopic` |
+| `messaging.gcp_pubsub.message.ordering_key` | string | The ordering key for a given message. If the attribute is not present, the message does not have an ordering key. | `ordering_key` |
```

Comment:
I have a question with is not-blocking to this PR:

there are similar concepts in different messaging systems, for example, AMQP has group-id property which is also called session-id in Azure ServiceBus, that identifies a set of messages to be delivered in order.

I wanted to check if the unification of such attribute would make any sense for GCP PubSub.

E.g. if we introduced something like `messaging.group.id`

**Pros:** Visualization tools/common dashboards/queries/etc can work better for all the systems that report it.
**Cons** Users would see an unfamiliar name and would need to discover how it maps to their messaging systems.

WDYT?

Reply:
I think that's a good point. 

They sound like they serve a similar purpose per service. Are they also optional for the other services?

I don't think it would be common to coordinate the same group name (group id/ordering key/session id) across the different messaging services though. So, I don't think it would help in the UI tools that are aggregating data/queries. 

Also, for Pub/Sub, I think it makes sense to have the attribute at the message level. Adding a group level might be easily confused with the `messaging.batch`.

I'd prefer to leave it as is. Resolving so I can submit. Re-open if you would like to discuss more.